README.md

```diff
@@ -13,68 +13,125 @@ npm install
 ```bash
 node run
 ```
 The command prompt `KnowSheet> ` will appear.
 
-... or evaluate one expression from the command-line:
+#### Evaluate from the command-line
 ```bash
 echo 'HTTP(GET("http://httpbin.org/get?query=1")).body' | node run
 ```
+If `stdin` is not a `tty`, the shell starts in a non-interactive mode.
 
-... or evaluate and pass the result to another command:
+#### Evaluate and pass the result to another command
 ```bash
 echo 'HTTP(GET("http://httpbin.org/get?query=1")).body' | node run | cat
 ```
+If `stdout` is not a `tty`, the result is emitted as text into `stdout`, the errors are emitted as text into `stderr`.
 
-Override the default prompt:
+#### Override the default prompt
 ```bash
 node run --prompt 'Bricks> '
 ```
 The command prompt `Bricks> ` will appear.
 
-### Expressions
+### HTTP client
 
-Perform an HTTP GET request and print the response:
+* Bricks provides utilities to perform HTTP requests.
+* By default, the redirects are forbidden and result in an error. They can be enabled per request.
+
+#### Perform an HTTP GET request
 ```
 KnowSheet> HTTP(GET("http://httpbin.org/get?query=1")).body
 ```
 
-Perform an HTTP POST request and print the response:
+#### Perform an HTTP POST request
 ```
 KnowSheet> HTTP(POST("http://httpbin.org/post", "BODY", "text/plain")).body
 ```
 
-Parse a JSON response into an object, get a field from it:
+#### Allow redirects
 ```
-KnowSheet> JSONParse(HTTP(GET("http://httpbin.org/get?query=1")).body).args
+KnowSheet> HTTP(GET("http://httpbin.org/redirect/5").AllowRedirects()).body
 ```
 
-POST a JSON-encoded object:
+#### Provide a custom `User-Agent`
 ```
-KnowSheet> JSONParse(HTTP(POST("http://httpbin.org/post", DemoObject())).body).json
+KnowSheet> HTTP(GET("http://httpbin.org/get").UserAgent("KnowSheet-REPL/1.0.0")).body
 ```
-<sup>The syntax mimics C++ Bricks exactly, as long as `DemoObject` is defined as a serializable type.</sup>
 
-Allow redirects:
+#### Provide extra HTTP headers
+For `GET` requests:
 ```
-KnowSheet> HTTP(GET("http://httpbin.org/redirect/5").AllowRedirects()).body
+KnowSheet> HTTP(GET("http://httpbin.org/get?query=1", HTTPHeaders().Set("Custom", "Header"))).body
 ```
 
-Provide a custom User-Agent:
+For `POST` requests:
 ```
-KnowSheet> HTTP(GET("http://httpbin.org/get").UserAgent("KnowSheet-REPL/1.0.0")).body
+KnowSheet> HTTP(POST("http://httpbin.org/post", "BODY", "text/plain", HTTPHeaders().Set("Custom", "Header"))).body
 ```
 
-Provide extra HTTP headers:
+### HTTP server
+
+* Bricks maintains a single server per port.
+* Each server accepts connections since start till the process exits.
+* There can only be a single handler for a given path.
+* The calls to the server methods can be chained.
+* The handler lambda syntax in the REPL mimics C++11 to a certain extent -- no full lambda support guaranteed.
+
+#### Start an HTTP server
 ```
-KnowSheet> HTTP(GET("http://httpbin.org/get?query=1", HTTPHeaders().Set("Custom", "Header"))).body
```

Comment:
Fixed in https://github.com/sompylasar/KnowSheet-REPL/commit/1abab3511848f6cd7c302855794fd0a2c0dd390c